Support for further encodings added

pjsharp opened the pull request:
- support for encodings iso-8859-[1..4] and freesat huffman added
- code improvements

pjsharp commented:
No longer relevant for PY2. Changes for PY3 are made in another fork.

pjsharp commented:
This code version is long obsolete. There is a newer one that doesn't need EitSupport and MetaSupport anymore. This is already merged in the current branch for PY3.

It's pointless to keep opening this PR and deleting my comments (also in other issues)!

dhwz commented:
Stop commeting on issues on this repo if you don't add changes here, this is the official EMC repository, it's not our fault the PY3 changes were not PRed, the repo is open to everyone.

pjsharp commented:
I was asked by jbleyel to look at the issues. The changes were merged in the fork which is currently used to build images (OATV 7.2) and used in the current feed

dhwz commented:
Right now you're just committing to a repo which doesn't keep compatibility with other images, by removing modules you already broke compatibility with others, main focus of EMC was always to make it run on any image.

pjsharp commented:
I didn't break any compatibility. I only removed code parts that are not necessary because they are natively supported by all E2 and now the reading of EventInfos will do like in all E2.

dhwz commented:
How can you tell it's working everywhere? Did you test it? On DreamOS, on older E2 forks?

pjsharp commented:
Not exactly with all of them. For example, I don't have a Dreambox. But the code parts I used now for EventInfo were already present in the existing code in different places before. There is actually nothing new inside, just unified, refactored and removed the unnecessary. If the code should no longer be compatible, then the compatibility was not given until now also.

And the issue area in the repo is common, other forks don't have their own issue area. So if you locked it, you take away the possibility for other users to be informed that there is a solution to their problems.

But whatever. I just wanted to help others.
